internal/controllers/machineset/machineset_controller.go

```
@@ -278,6 +279,15 @@ func (r *Reconciler) reconcile(ctx context.Context, cluster *clusterv1.Cluster,
 			r.recorder.Eventf(machineSet, corev1.EventTypeNormal, "SuccessfulAdopt", "Adopted Machine %q", machine.Name)
 		}
 
+		// Ensure that the version of the controllerReference is up-to-date.
+		if metav1.GetControllerOf(machine).APIVersion != machineSet.APIVersion {
+			err := r.updateControllerReference(ctx, machineSet, machine)
+			if err != nil {
+				log.Error(err, "Failed to update Machine controller reference.")
```

I would also return the error here